packages/api-client-core/src/operationRunners.ts

```diff
@@ -29,7 +29,7 @@ export const findOneRunner = async <Shape extends RecordShape = any>(
   const response = await modelManager.connection.currentClient.query(plan.query, plan.variables).toPromise();
   const assertSuccess = throwOnEmptyData ? assertOperationSuccess : assertNullableOperationSuccess;
   const record = assertSuccess(response, [operation]);
-  return record ? hydrateRecord<Shape>(response, record) : null;
+  return hydrateRecord<Shape>(response, record);
```

Comment:
Just a refresher, but `hydrateRecord` is okay with the possibility that record could be `null`, eh?

Reply:
Yeah, that's actually why I missed it at first: it does an `Object.assign(defaultObject, incomingData)`, type of thing, so you end up with just the default object.

In practical terms, it will either throw an error, or end up with a null return, so there shouldn't really be any instances where that "empty" object gets used.